OpenRA.Game/Traits/World/Shroud.cs

```diff
@@ -163,42 +162,105 @@ void UpdateVisibility(Actor a, ref CPos[] visible)
 			AddVisibility(a, ref visible);
 		}
 
-		void AddShroudGeneration(Actor a, ref CPos[] shrouded)
+		void AddDisruptionGenerator(Actor a, ref CPos[] underDisruption)
+		{
+			var cd = a.TraitOrDefault<CreatesDisruptionField>();
+			if (cd == null || a.Owner.IsAlliedWith(self.Owner) || cd.Range == WRange.Zero || !self.World.LobbyInfo.GlobalSettings.Fog)
```

Comment:
Isn't the check for the disabled fog also done at the trait level? Do we need to duplicate it here?